src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -0,0 +1,188 @@
+            try {
+                /** @var GenerateMailTemplatesService $generator */
+                $generator = $this->get('prestashop.service.generate_mail_templates');
+                //Overwrite theme folder if selected
+                if (!empty($data['theme'])) {
+                    $themeFolder = $this->getParameter('themes_dir') . $data['theme'];
+                    $generator
+                        ->setCoreMailsFolder($themeFolder . '/mails')
+                        ->setModulesMailFolder($themeFolder . '/modules')
+                    ;
+                }
+
+                $generator->generateMailTemplates($data['mail_theme'], $data['language'], $data['overwrite']);
+
+                $flashMessage = 'Successfully generated mail templates for theme %s with locale %s';
+                if ($data['overwrite']) {
+                    $flashMessage = 'Successfully overrode mail templates for theme %s with locale %s';
+                }
+                $this->addFlash(
+                    'success',
+                    $this->trans(
+                        sprintf(
+                            $flashMessage,
+                            $data['mail_theme'],
+                            $data['language']
+                        ),
+                        'Admin.Notifications.Success'
+                    )
+                );
+            } catch (CoreException $e) {
+                $this->flashErrors([
+                    $this->trans(
+                        sprintf(
+                            'Could not generate mail templates for theme %s with locale %s',
+                            $data['mail_theme'],
+                            $data['language']
+                        ),
+                        'Admin.Notifications.Error'
+                    ),
+                    $e->getMessage(),
+                ]);
+            }
+        }
+
+        return $this->redirectToRoute('admin_mail_theme_index');
+    }
+
+    /**
+     * Save mail theme configuration
+     *
+     * @AdminSecurity("is_granted(['update'], request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function saveConfigurationAction(Request $request)
+    {
+        /** @var FormHandlerInterface $formHandler */
+        $formHandler = $this->getMailThemeFormHandler();
+        /** @var Form $form */
+        $form = $formHandler->getForm()->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            if (!$form->isValid()) {
+                $this->flashErrors($this->getFormErrorsForJS($form));
+
+                return $this->redirectToRoute('admin_mail_theme_index');
+            }
+
+            $errors = $formHandler->save($form->getData());
+            if (empty($errors)) {
+                $this->addFlash(
+                    'success',
+                    $this->trans(
+                        'Successfully saved mail theme configuration.',
```

Comment:
Possible to write _Mail theme configuration saved successfully_ instead?

Reply:
done